crates/arroyo-connectors/src/nats/mod.rs

```diff
@@ -0,0 +1,316 @@
+                format!("NatsSink<{:?}>", stream_or_subject.clone()),
+            ),
+        };
+
+        let schema = schema
+            .map(|s| s.to_owned())
+            .ok_or_else(|| anyhow!("No schema defined for NATS connection"))?;
+
+        let format = schema
+            .format
+            .as_ref()
+            .map(|t| t.to_owned())
+            .ok_or_else(|| anyhow!("'format' must be set for NATS connection"))?;
+
+        let config = OperatorConfig {
+            connection: serde_json::to_value(config).unwrap(),
+            table: serde_json::to_value(table).unwrap(),
+            rate_limit: None,
+            format: Some(format),
+            bad_data: schema.bad_data.clone(),
+            framing: schema.framing.clone(),
+        };
+
+        Ok(Connection {
+            id,
+            connector: self.name(),
+            name: name.to_string(),
+            connection_type: connection_type,
+            schema,
+            config: serde_json::to_string(&config).unwrap(),
+            description: desc,
+        })
+    }
+
+    fn from_options(
+        &self,
+        name: &str,
+        options: &mut HashMap<String, String>,
+        schema: Option<&ConnectionSchema>,
+        profile: Option<&ConnectionProfile>,
+    ) -> anyhow::Result<Connection> {
+        let connection = profile
+            .map(|p| {
+                serde_json::from_value(p.config.clone()).map_err(|e| {
+                    anyhow!("Invalid config for profile '{}' in database: {}", p.id, e)
+                })
+            })
+            .unwrap_or_else(|| Self::connection_from_options(options))?;
+
+        let table = Self::table_from_options(options)?;
+
+        Self::from_config(&self, None, name, connection, table, schema)
+    }
+
+    fn make_operator(
+        &self,
+        profile: Self::ProfileT,
+        table: Self::TableT,
+        config: OperatorConfig,
+    ) -> anyhow::Result<OperatorNode> {
+        Ok(match table.connector_type {
+            ConnectorType::Source { .. } => OperatorNode::from_source(Box::new(NatsSourceFunc {
+                stream: table.client_configs.get("nats.stream").cloned().unwrap(),
+                servers: profile.servers.clone(),
+                connection: profile.clone(),
+                table: table.clone(),
+                framing: config.framing,
+                format: config.format.unwrap(),
+                bad_data: config.bad_data,
+                consumer_config: get_nats_client_config(&profile, &table),
+                messages_per_second: NonZeroU32::new(
+                    config
+                        .rate_limit
+                        .map(|l| l.messages_per_second)
+                        .unwrap_or(u32::MAX),
+                )
+                .unwrap(),
+            })),
+            ConnectorType::Sink { .. } => OperatorNode::from_operator(Box::new(NatsSinkFunc {
+                subject: table.client_configs.get("nats.subject").cloned().unwrap(),
```

Comment:
Similarly here, we probably want to use the subject in `ConnectorType::Sink { subject }`

Reply:
Done 👍